### ISSUE TITLE
OpenAPI 3.x YAML schemas not being parsed correctly with swagger-codegen 3.0.21

### ISSUE THREAD

**Issue by alebcay** [swagger-api/swagger-codegen#10415]
##### Description

At Homebrew, we're currently working on updating our packaged version of `swagger-codegen` from 3.0.20 to 3.0.21. Part of our upgrade process tests the newly built generator JAR with generating for `html` language from a trivial example OpenAPI 3.x YAML snippet. This error was found during the testing process.

Homebrew issue: https://github.com/Homebrew/homebrew-core/pull/58797

##### Swagger-codegen version

3.0.21. We ran the same test during the upgrade process for 3.0.20 and did not experience this issue.

##### Swagger declaration file content or url

```yaml
---
openapi: 3.0.0
info:
  version: 0.0.0
  title: Simple API
paths:
  /:
    get:
      responses:
        200:
          description: OK
```

I've also tested the same declaration sans the three dashes on top and also with the [this `petstore.yaml`](https://raw.githubusercontent.com/OpenAPITools/openapi-generator/master/modules/openapi-generator/src/test/resources/3_0/petstore.yaml) with similar/same error.

The error output:
```
[main] INFO io.swagger.parser.Swagger20Parser - reading from minimal.yaml
[main] INFO io.swagger.parser.Swagger20Parser - reading from minimal.yaml
[main] ERROR io.swagger.parser.SwaggerCompatConverter - failed to read resource listing
com.fasterxml.jackson.core.JsonParseException: Unexpected character ('-' (code 45)) in numeric value: expected digit (0-9) to follow minus sign, for valid numeric value
 at [Source: (String)"---
openapi: 3.0.0
info:
  version: 0.0.0
  title: Simple API
paths:
  /:
    get:
      responses:
        200:
          description: OK
"; line: 1, column: 3]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1840)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:712)
	at com.fasterxml.jackson.core.base.ParserMinimalBase.reportUnexpectedNumberChar(ParserMinimalBase.java:541)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleInvalidNumberStart(ReaderBasedJsonParser.java:1652)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._parseNegNumber(ReaderBasedJsonParser.java:1406)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:758)
	at com.fasterxml.jackson.databind.ObjectMapper._readTreeAndClose(ObjectMapper.java:4231)
	at com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:2711)
	at io.swagger.parser.SwaggerCompatConverter.readResourceListing(SwaggerCompatConverter.java:210)
	at io.swagger.parser.SwaggerCompatConverter.read(SwaggerCompatConverter.java:123)
	at io.swagger.parser.SwaggerParser.read(SwaggerParser.java:83)
	at io.swagger.codegen.config.CodegenConfigurator.toClientOptInput(CodegenConfigurator.java:437)
	at io.swagger.codegen.cmd.Generate.run(Generate.java:297)
	at io.swagger.codegen.SwaggerCodegen.main(SwaggerCodegen.java:35)
[main] INFO io.swagger.codegen.ignore.CodegenIgnoreProcessor - No .swagger-codegen-ignore file found.
Exception in thread "main" java.lang.RuntimeException: missing swagger input or config!
	at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:763)
	at io.swagger.codegen.cmd.Generate.run(Generate.java:299)
	at io.swagger.codegen.SwaggerCodegen.main(SwaggerCodegen.java:35)
```

##### Command line used for generation

```
java -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate -i minimal.yaml -l html
```

```
openjdk version "14.0.1" 2020-04-14
OpenJDK Runtime Environment (build 14.0.1+14)
OpenJDK 64-Bit Server VM (build 14.0.1+14, mixed mode, sharing)
```

(The CI test ran on macOS's bundled Java; the openjdk version is on my local machine)

##### Steps to reproduce

1. Build package with `mvn clean package`.
2. Place the declaration in `minimal.yaml`.
3. Run the above mentioned command.

##### Related issues/PRs

https://github.com/swagger-api/swagger-codegen/issues/478

##### Suggest a fix/enhancement

It seems to be trying to parse the YAML as JSON, so I would consider that to be a starting point for investigating further.

**Comment by hparadiz** [comment]
Same issue 

```
$ ./build-js.sh [main] INFO io.swagger.parser.Swagger20Parser - reading from ../incentfit-website/webapp/swagger-api.yaml
[main] INFO io.swagger.parser.Swagger20Parser - reading from ../incentfit-website/webapp/swagger-api.yaml
[main] ERROR io.swagger.parser.SwaggerCompatConverter - failed to read resource listing
com.fasterxml.jackson.core.JsonParseException: Unexpected character ('-' (code 45)) in numeric value: expected digit (0-9) to follow minus sign, for valid numeric value
 at [Source: (StringReader); line: 1, column: 3]
        at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1840)
        at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:712)
        at com.fasterxml.jackson.core.base.ParserMinimalBase.reportUnexpectedNumberChar(ParserMinimalBase.java:541)
        at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleInvalidNumberStart(ReaderBasedJsonParser.java:1652)
        at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._parseNegNumber(ReaderBasedJsonParser.java:1406)
        at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:758)
        at com.fasterxml.jackson.databind.ObjectMapper._readTreeAndClose(ObjectMapper.java:4231)
        at com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:2711)
        at io.swagger.parser.SwaggerCompatConverter.readResourceListing(SwaggerCompatConverter.java:210)
        at io.swagger.parser.SwaggerCompatConverter.read(SwaggerCompatConverter.java:123)
        at io.swagger.parser.SwaggerParser.read(SwaggerParser.java:83)
        at io.swagger.codegen.config.CodegenConfigurator.toClientOptInput(CodegenConfigurator.java:437)
        at io.swagger.codegen.cmd.Generate.run(Generate.java:297)
        at io.swagger.codegen.SwaggerCodegen.main(SwaggerCodegen.java:35)
Exception in thread "main" java.lang.RuntimeException: missing swagger input or config!
        at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:763)
        at io.swagger.codegen.cmd.Generate.run(Generate.java:299)
        at io.swagger.codegen.SwaggerCodegen.main(SwaggerCodegen.java:35)
```

**Comment by ruud-de-jong** [comment]
I think I have a related issue with Maven Java and Swagger upgrade from 3.0.20 to 3.0.21 based on Open API 3.x.
In my case, I have a POST with x-www-form-urlencoded.
When upgrading, all my parameters disappear.

`    post:
      operationId: postForm
      requestBody:
        required: true
        content:
          application/x-www-form-urlencoded:
            schema:
              type: object
              properties:
                param_1:
                  type: "string"
                param_2:
                  type: "string"
`
so it now becomes a no-args method.

**Comment by alebcay** [comment]
This seems to be resolved with version 3.0.22.
